src/box/txn_limbo.c

```diff
@@ -40,6 +40,69 @@
 
 struct txn_limbo txn_limbo;
 
+static int
+synchro_request_write_without_error_handling(const struct synchro_request *req);
+
+static void
+txn_limbo_read_confirm(struct txn_limbo *limbo, int64_t lsn);
+
+/**
+ * Write a confirmation entry to the WAL. After it's written all the
+ * transactions waiting for confirmation may be finished.
+ */
+static int
+txn_limbo_write_confirm(struct txn_limbo *limbo)
+{
+	assert(!limbo->is_in_rollback);
+	struct synchro_request req = {
+		.type = IPROTO_RAFT_CONFIRM,
+		.replica_id = limbo->owner_id,
+		.lsn = limbo->volatile_confirmed_lsn,
+		.term = 0,
+		.confirmed_vclock = NULL,
+	};
+	return synchro_request_write_without_error_handling(&req);
+}
+
+static int
+txn_limbo_worker_bump_confirmed_lsn(struct txn_limbo *limbo)
+{
+	assert(limbo->is_in_rollback ||
+	       limbo->volatile_confirmed_lsn >= limbo->confirmed_lsn);
+	if (!txn_limbo_is_owned_by_current_instance(limbo) ||
+	    limbo->volatile_confirmed_lsn == limbo->confirmed_lsn)
+		return 0;
+	if (limbo->is_in_rollback)
+		return -1;
+	if (txn_limbo_write_confirm(limbo) != 0) {
+		diag_log();
+		return -1;
+	}
+
+	limbo->confirmed_lsn = limbo->volatile_confirmed_lsn;
+	vclock_follow(&limbo->confirmed_vclock, limbo->owner_id,
+		      limbo->confirmed_lsn);
+	txn_limbo_read_confirm(limbo, limbo->confirmed_lsn);
```

Comment:
Why do we need assigning `confirmed_lsn` value and bumping `confirmed_vclock` here, when it's all done in `txn_limbo_read_confirm`. We just need to pass value of `volatile_confirmed_lsn` before `journal_write` to it

Reply:
I overlooked it, thanks!